ext/TensorAlgebraBlockSparseArraysGradedUnitRangesExt/TensorAlgebraBlockSparseArraysGradedUnitRangesExt.jl

```diff
@@ -0,0 +1,33 @@
+module TensorAlgebraBlockSparseArraysGradedUnitRangesExt
+
+using BlockArrays: Block, blocksize
+using BlockSparseArrays: BlockSparseMatrix, @view!
+using GradedUnitRanges: AbstractGradedUnitRange, dual, tensor_product
+using Random: AbstractRNG
+using TensorAlgebra: TensorAlgebra, random_unitary!
+
+function TensorAlgebra.:⊗(a1::AbstractGradedUnitRange, a2::AbstractGradedUnitRange)
+  return tensor_product(a1, a2)
+end
+
+function TensorAlgebra.square_zero_map(
+  elt::Type, ax::Tuple{AbstractGradedUnitRange,Vararg{AbstractGradedUnitRange}}
+)
+  return BlockSparseArray{elt}(undef, (dual.(ax)..., ax...))
+end
+
+function TensorAlgebra.random_unitary!(
+  rng::AbstractRNG,
+  a::BlockSparseMatrix{
+    <:Any,<:Any,<:Any,<:Tuple{AbstractGradedUnitRange,Vararg{AbstractGradedUnitRange}}
+  },
+)
+  # TODO: Define and use `blockdiagindices`
```

Comment:
Something worth considering: in this function we are not guaranteed the user inputted a square matrix, in which case this function would still work, but spit out a random left- or right- isometry. I think if we call the function `unitary`, we probably want to check for this, either by an explicit `checksquare`, or some other means.

Reply:
Good point, I had that in mind but forgot to add that check. We should definitely check it is square (and more specifically, the blocks are the same and the codomain is the dual of the domain).

Reply:
After adding this check, I realized there is a crucial thing missing from this PR, which is specifying that the codomain and domain are duals of each other. That will require a separate PR to `fusedims` to support that, though @ogauthe is reworking `fusedims`/`splitdims` to accommodate FusionTensors so I'll hold off on that for now.